storybook/pages/ChatViewPocPage.qml

```diff
@@ -0,0 +1,270 @@
+        }
+
+        return result.join("");
+    }
+
+    function generateSampleModelData(size, maxWordCount) {
+        const modelData = [];
+
+        function randomDate(start, end) {
+          return new Date(start.getTime() + Math.random() * (end.getTime() - start.getTime()))
+        }
+
+        modelData.push({ text: "**HELLO WORLD!** (This is very first message)", images: [], date: new Date() })
+
+        for (let i = 0; i < size; i++) {
+            const wordCount = Math.floor(Math.random() * maxWordCount) + 1; // 1 to maxCount inclusive
+
+            const text = generateLoremIpsum(wordCount)
+            const date = randomDate(new Date(2012, 0, 1), new Date())
+            const avatar = `https://picsum.photos/id/${(i) % 70}/50/50`
+
+            const imageCount = Math.round(Math.random() - 0.4) * Math.floor(Math.random() * 10)
+            const imagesSeed = Math.round(Math.random() * 100)
+            const images = []
+
+            for (let j = 0; j < imageCount; j++)
+                images.push({url: `https://picsum.photos/id/${(imagesSeed + j) % 70}/1200/1300`})
+
+            modelData.push({ text, images, date, avatar })
+        }
+
+        modelData.push({ text: "**GOOD BYE!** (This is the latest message)", images: [], date: new Date() })
+
+        return modelData
+    }
+
+    function generatePlaceholderContent(size = 12, minLength = 30, maxLength = 140,
+                                        minCount = 1, maxCount = 15) {
+
+        const array = []
+
+        function getRandomInt(max) {
+          return Math.floor(Math.random() * max);
+        }
+
+        for (let i = 0; i < size; i++) {
+            const count = getRandomInt(maxCount - minCount) + minCount
+
+            const subarray = []
+
+            for (let j = 0; j < count; j++) {
+                subarray.push(getRandomInt(maxLength - minLength) + minLength)
+            }
+
+            array.push(subarray)
+        }
+
+        return array
+    }
+
+    Rectangle {
+        anchors.fill: parent
+
+        color: "#232325"
+    }
+
+    ListModel {
+        id: lm
+
+        Component.onCompleted: {
+            const size = 320
+            const maxWordCount = 40
+
+            append(root.generateSampleModelData(size, maxWordCount))
+        }
+    }
+
+    Flickable {
+        id: flickable
+
+        ScrollBar.vertical: ScrollBar {
+            id: scrollbar
+
+            onPressedChanged: {
+                if (pressed)
+                    return
+
+                const isTopPlaceholderVisible = flickable.contentY < topPlaceholder.height
+
+                if (isTopPlaceholderVisible) {
+                    const first = messagesRepeater.itemAt(0)
+
+                    const offset = first.y - flickable.contentY
+
+                    flickable.contentY = Qt.binding(() => {
+                        return first.y - offset
+                    })
+
+                    const shift = Math.min(40, indexFilter.minimumIndex)
```

Comment:
I guess this should be called when scrolling by wheel, trackpad or touch as well